core/src/commonMain/kotlin/dev/schlaubi/lavakord/audio/internal/NodeImpl.kt

```diff
@@ -118,30 +119,37 @@ internal class NodeImpl(
 
         retry.reset()
 
-        LOG.debug { "Successfully connected to node: $name ($host)" }
+        LOG.info { "Successfully connected to node: $name ($host)" }
 
         while (!session.incoming.isClosedForReceive) {
             try {
                 onEvent(session.receiveDeserialized())
-            } catch (e: WebsocketDeserializeException) {
-                LOG.warn(e) { "An error occurred whilst decoding incoming websocket packet" }
+            } catch (e: ClosedReceiveChannelException) {
+                break
+            } catch (e: Exception) {
+                LOG.warn(e) { "An exception occurred whilst decoding incoming websocket packet" }
             }
         }
-        val reason = session.closeReason.await()
-        if (reason?.knownReason == CloseReason.Codes.NORMAL) return
+
         available = false
-        LOG.warn { "Disconnected from websocket for: $reason. Music will continue playing if we can reconnect within the next $resumeTimeout seconds" }
-        reconnect(resume = true)
+        val reason = session.closeReason.await()
+        val resumeAgain = resume && reason?.knownReason != CloseReason.Codes.NORMAL
+        if (resumeAgain) {
+            LOG.warn { "Disconnected from websocket for: $reason. Music will continue playing if we can reconnect within the next $resumeTimeout seconds" }
+        } else {
+            LOG.warn { "Disconnected from websocket for: $reason. Not resuming." }
+        }
+        reconnect(resume = resumeAgain)
     }
 
     private suspend fun reconnect(e: Throwable? = null, resume: Boolean = false) {
-        LOG.error(e) { "Exception whilst trying to connect. Reconnecting" }
         if (retry.hasNext) {
+            LOG.error { "Exception whilst trying to connect: '${e?.message}'. Reconnecting" }
             retry.retry()
             connect(resume)
         } else {
             lavakord.removeNode(this)
-            error("Could not reconnect to websocket after to many attempts")
+            throw RuntimeException("Could not reconnect to websocket after to many attempts", e)
```

Comment:
```suggestion
            throw IllegalStateException("Could not reconnect to websocket after to many attempts", e)
```

Reply:
I don't think that IllegalStateException is a meaningful exception name to communicate to the user.

From the Java Javadoc:

> Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.

There's a typo in the error message that I'll commit a fix for

Reply:
I am not sure which exception to use, however, I don't think that the top-level RuntimeException should be used here, an IllegalStateException would already be more meaningful, however, I am open to suggestions

Reply:
It's very common to throw around a `RuntimeException` when there isn't an exception that better fits. Though if you want to use `IllegalStateException` then I don't particularly mind.

Reply:
I am still in favor of an IllegalStateException